spec/features/rooms/index_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'index view' do
+  let!(:holliday) { Hotel.create!(name: 'Holliday Inn', location: 'Buxton') }
+  let!(:harbor) { Hotel.create!(name: 'Portland Harbor', location: 'Portland') }
+  let!(:room_1) { harbor.rooms.create!(rate: 180, suite: "Presidential")}
+  let!(:room_2) { harbor.rooms.create!(rate: 185, suite: "Vice-Presidential")}
+  let!(:room_3) { holliday.rooms.create!(rate: 160, suite: "Honeymoon")}
+  let!(:room_4) { holliday.rooms.create!(rate: 10, suite: "Broom Closet")}
+  let!(:steve) { room_1.guests.create!(name: 'Steve', nights: 3) }
+  let!(:dave) { room_2.guests.create!(name: 'Dave', nights: 5) }
+  let!(:jim) { room_3.guests.create!(name: 'Jim', nights: 1)}
+
+  it 'lists all rooms and associated attributes' do
+    visit '/rooms'
+
+    GuestRoom.create!(guest: steve, room: room_1)
+    GuestRoom.create!(guest: dave, room: room_1)
+
+    GuestRoom.create!(guest: steve, room: room_2)
+
+    GuestRoom.create!(guest: dave, room: room_3)
+
+    GuestRoom.create!(guest: dave, room: room_4)
+
+    expect(page).to have_content(room_1.suite)
+    expect(page).to have_content(room_1.rate)
+    expect(page).to have_content(harbor.name)
+    expect(page).to have_content(room_1.guests.count)
+
+    
+    expect(page).to have_content(room_2.suite)
+    expect(page).to have_content(room_2.rate)
+    expect(page).to have_content(room_2.guests.count)
+
+    expect(page).to have_content(room_3.suite)
+    expect(page).to have_content(room_3.rate)
+    expect(page).to have_content(holliday.name)
+    expect(page).to have_content(room_3.guests.count)
+  end
```

Comment:
Like I mentioned in your previous test, interpolation tests are better and let's say you couldn't properly do some interpolation in the assertions. You could then use some `within` blocks. The most important thing is that we want to be absolutely sure that the things were are asserting to be true are actually true and not false-positive.

```ruby
within "#room-#{@room_1.id}" do
  expect(page).to have_content(@room1.suite)
  expect(page).to have_content(@room1.rate)
  expect(page).to have_content(@hotel1.name)
end
```

This would mean that you would need to update your view to something like this:

```ruby
<% @rooms.each do |room| %>
  <div id="room-<%= room.id %>">
    <h1><%= room.suite %></h1>
    <p>Rate: <%= room.rate %></p>
    <p>Hotel: <%= room.hotel.name %>
    <p>Rented: <%= room.guests.count %></p>
  </div>
<% end %>
```